src/main/java/org/gooteam/balance/BalanceTicker.java

```diff
@@ -0,0 +1,70 @@
+package org.gooteam.balance;
+
+import org.gooteam.BalanceRepository;
+import org.gooteam.event.GooMessage;
+import org.gooteam.listener.GooMessageListener;
+import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Arrays;
+import java.util.HashSet;
+
+public class BalanceTicker implements GooMessageListener {
+    private static final Logger logger = LoggerFactory.getLogger(GooMessageListener.class);
+
+    private static final double BASE_GOO_WEIGHT = 0.01;
+
+    private static final HashSet<String> SLIME_STRINGS = new HashSet<>(
+        Arrays.asList(
+            "slime",
+            "slimy",
+            "goop",
+            "squelch",
+            "bubble",
+            "splat"
+        )
+    );
+
+    private final BalanceRepository repo;
+
+    public BalanceTicker(BalanceRepository repo) {
+        this.repo = repo;
+    }
+
+    @Override
+    public void onMessage(@NotNull GooMessage message) {
+        logger.info("running balance ticker functionality!!!");
+        String userId = message.getEvent().getAuthor().getId();
+        double balanceDelta = computeMessageWeight(message.getContent());
+        repo.changeBalance(
+                message.getEvent().getAuthor().getId(),
+                balanceDelta
+        );
+
+
+        logger.info("uptick balance for ID {}: {} cm3", userId, balanceDelta);
+    }
+
+    private double computeMessageWeight(String message) {
+        // goes nuts for links
+        double lengthWeight = Math.clamp(Math.log(Math.max(message.length(), 1)), 0.1, 8.0);
```

Comment:
Is there a way to not count links towards the lengthWeight? or maybe give them the same weight as like a 4 letter word

Reply:
that makes sense - i think there's a link parser that'll work in java?

Reply:
went w using a dumb regex expression to remove http/https links